seacrowd/sea_datasets/parallel_asian_treebank/parallel_asian_treebank.py

```diff
@@ -0,0 +1,167 @@
+}
+"""
+
+_DESCRIPTION = """\
+The ALT project aims to advance the state-of-the-art Asian natural language processing (NLP) techniques through the open collaboration for developing and using ALT. 
+It was first conducted by NICT and UCSY as described in Ye Kyaw Thu, Win Pa Pa, Masao Utiyama, Andrew Finch and Eiichiro Sumita (2016). 
+Then, it was developed under ASEAN IVO. 
+The process of building ALT began with sampling about 20,000 sentences from English Wikinews, and then these sentences were translated into the other languages. 
+ALT now has 13 languages: Bengali, English, Filipino, Hindi, Bahasa Indonesia, Japanese, Khmer, Lao, Malay, Myanmar (Burmese), Thai, Vietnamese, Chinese (Simplified Chinese).
+"""
+
+_HOMEPAGE = "https://www2.nict.go.jp/astrec-att/member/mutiyama/ALT/"
+
+_LICENSE = Licenses.CC_BY_4_0.value
+
+_URL = "https://www2.nict.go.jp/astrec-att/member/mutiyama/ALT/ALT-Parallel-Corpus-20191206.zip"
+
+_SUPPORTED_TASKS = [Tasks.MACHINE_TRANSLATION]
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class ParallelAsianTreebank(datasets.GeneratorBasedBuilder):
+    """The ALT project aims to advance the state-of-the-art Asian natural language processing (NLP) techniques through the open collaboration for developing and using ALT"""
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=_SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_t2t",
+            version=_SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_t2t",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self):
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "text_1": datasets.Value("string"),
+                    "text_2": datasets.Value("string"),
+                    "text_1_name": datasets.Value("string"),
+                    "text_2_name": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "seacrowd_t2t":
+            features = schemas.text2text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        if self.config.data_dir is None:
+            raise ValueError(
+                "This is a local dataset. Please pass the data_dir kwarg to load_dataset."
+            )
+        else:
+            data_dir = self.config.data_dir
```

Comment:
I'm wondering about the reason why changed this to be a local dataset, is it that you want to support the splits? The split files can also be downloaded. Find ALT-Standard-Split on the page? https://www2.nict.go.jp/astrec-att/member/mutiyama/ALT/

Reply:
But this just contains the link of wikipedia. And should be preprocessed first as mentioned in here

![image](https://github.com/SEACrowd/seacrowd-datahub/assets/11547086/a545b295-7a62-4ea6-9c3e-e8cf51009ffc)

Anw. I implemented the dataloader for local dataset as discussed with kak @SamuelCahyawijaya and he had confirmed and approved it. I guess other reviewers had been aware about it.

Reply:
In README.txt, the author describes how the statement identifier contains the URLID which you can use to decide the split for each line. @SamuelCahyawijaya what is the reason why this has to be implemented as local dataset?

```
Each text file is the following format.

SNT.URLID.SNTID

The website for URLID is listed in URL.txt
SNTID is from 1 to 20000. Some sentences divided into two are
represented by xxx and xxx-1. For example,

SNT.114026.167	"I'm the only candidate who lives here with roots here." The Teamster's lawyer is originally from Edmonton, but has been living in Vancouver environs since 1991.
SNT.114026.167-1	The Teamster's lawyer is originally from Edmonton, but has been living in Vancouver environs since 1991.
```

Reply:
Yeah I think this can be done without having the files locally. I'll take a stab